stdlib/public/core/KeyValuePairs.swift

```diff
@@ -125,6 +125,56 @@ extension KeyValuePairs: RandomAccessCollection {
   }
 }
 
+extension KeyValuePairs: Equatable where Key: Equatable, Value: Equatable {
+    /// Returns a Boolean value indicating whether two `KeyValuePairs` instances
+    /// contain the same elements in the same order.
+    ///
+    /// You can use the equal-to operator (`==`) to compare any two `KeyValuePairs` instances
+    /// that store the same, `Equatable`-conforming `Key` and `Value` types.
+    ///
+    /// - Parameters:
+    ///   - lhs: An `KeyValuePairs` instance to compare.
+    ///   - rhs: Another `KeyValuePairs` instance to compare.
+    @inlinable
+    public static func ==(lhs: KeyValuePairs<Key, Value>, rhs: KeyValuePairs<Key, Value>) -> Bool {
```

Comment:
Modeled after the existing [`Array.==(lhs:rhs:)` implementation](https://github.com/apple/swift/blob/516ad28c504d835e4c189930020016a96d72bd06/stdlib/public/core/Array.swift#L1730)